modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/consensus/ConsensusManager.scala

```diff
@@ -159,19 +157,28 @@ object ConsensusManager {
           case Some(state) =>
             state.status match {
               case Finished(signedArtifact, majorityTrigger) =>
-                Metrics[F].recordTime("dag_consensus_duration", state.statusUpdatedAt.minus(state.createdAt)) >>
+                Clock[F].monotonic.flatMap { finishedAt =>
+                  Metrics[F].recordTime("dag_consensus_duration", finishedAt - state.createdAt)
+                } >>
+                  deregisterPeers(key, state.removedFacilitators) >>
                   consensusStorage
                     .tryUpdateLastKeyAndArtifactWithCleanup(state.lastKey, key, signedArtifact)
                     .ifM(
                       afterConsensusFinish(majorityTrigger),
                       logger.info("Skip triggering another consensus")
                     ) >> nodeStorage.tryModifyStateGetResult(Observing, Ready).void
               case _ =>
-                internalCheckForStateUpdate(key, resources)
+                stallDetection(key, state) >>
```

Comment:
Improvement for the future already discussed offline - I propose running `stallDetection` only once for every stage as only the first run is useful the rest is bound to fail. So I guess we want the `stallDetection` background task to start in right after status switch (starting next stage).

Reply:
Agree, I think it's an important improvement because we want to keep the "flow" clean and under control. It will require returning `Ior` instead of `Maybe` from the `ConsensusStateUpdater`. I will create a ticket for this and fix it as a part of that ticket if you don't mind.